services/app-api/src/postgres/contractAndRates/insertRate.ts

```diff
@@ -59,28 +125,30 @@ async function insertDraftRate(
                 },
             })
 
-            return {
+            const finalRate: RateType = {
                 id: rate.id,
+                status: 'DRAFT',
```

Comment:
I think you mentioned using `getContractStatus` here instead of hard coding the status.